qiskit/providers/ibmq/utils/converters.py

```diff
@@ -36,12 +39,63 @@ def utc_to_local(utc_dt: Union[datetime.datetime, str]) -> datetime.datetime:
     if isinstance(utc_dt, str):
         utc_dt = dateutil.parser.parse(utc_dt)
     if not isinstance(utc_dt, datetime.datetime):
-        raise TypeError('Input is not string or datetime.')
+        raise TypeError('Input `utc_dt` is not string or datetime.')
     utc_dt = utc_dt.replace(tzinfo=datetime.timezone.utc)  # type: ignore[arg-type]
     local_dt = utc_dt.astimezone(tz.tzlocal())  # type: ignore[attr-defined]
     return local_dt
 
 
+def local_to_utc(local_dt: Union[datetime.datetime, str]) -> datetime.datetime:
+    """Convert a local ``datetime`` object or string to a UTC ``datetime``.
+
+    Args:
+        local_dt: Input local `datetime` or string.
+
+    Returns:
+        A ``datetime`` in UTC.
+
+    Raises:
+        TypeError: If the input parameter value is not valid.
+    """
+    if isinstance(local_dt, str):
+        local_dt = dateutil.parser.parse(local_dt)
+    if not isinstance(local_dt, datetime.datetime):
+        raise TypeError('Input `local_dt` is not string or datetime.')
+    utc_dt = local_dt.astimezone(tz.UTC)
+    return utc_dt
+
+
+def datetime_to_str(date_time: datetime.datetime) -> str:
+    """Convert a datetime object to a formatted string representation.
+
+    Note:
+        The `date_time` is converted into a string representation based
+        on its timezone info. If the `date_time` does not have timezone
+        info, it is considered to be in local time.
+
+    Args:
+        date_time: Input `datetime` object.
+
+    Returns:
+        A string representation of the `datetime`.
+    """
+    if date_time.tzinfo in (datetime.timezone.utc, tz.UTC):
+        return date_time.strftime(UTC_DATETIME_TO_STR_FORMATTER)
+    return date_time.strftime(LOCAL_DATETIME_TO_STR_FORMATTER)
```

Comment:
Any reason you use your own formatting rather than `isoformat()` ?

Reply:
At first, I used a formatter since there was a specific format the time data was returned in. So, this function was meant to do the reverse. But, we use `dateutil.parser` to handle conversions, which allows date time conversion to be more flexible. So, it would be better to use `isoformat()`, as was used before 👍